cluster/dragon/integration/test_cluster.go

```diff
@@ -0,0 +1,100 @@
+package integration
+
+import (
+	"time"
+
+	"github.com/squareup/pranadb/cluster"
+	"github.com/squareup/pranadb/cluster/dragon"
+	"github.com/squareup/pranadb/common"
+	"github.com/squareup/pranadb/conf"
+	"github.com/squareup/pranadb/errors"
+)
+
+// TestCluster is a DragonBoat cluster for integration testing.
+type TestCluster struct {
+	dataDir           string
+	numShards         int
+	replicationFactor int
+	tlsConfig         conf.TLSConfig
+	nodes             []*dragon.Dragon
+}
+
+// NewTestCluster creates a new DragonBoat integration test cluster.
+func NewTestCluster(dataDir string, numShards int, replicationFactor int, tlsConfig conf.TLSConfig) *TestCluster {
+	return &TestCluster{
+		dataDir:           dataDir,
+		numShards:         numShards,
+		replicationFactor: replicationFactor,
+		tlsConfig:         tlsConfig,
+	}
+}
+
+// Start starts a DragonBoat cluster for integration testing
+func (c *TestCluster) Start() error {
```

Comment:
There's already a method for starting a cluster in dragon_integration_test.go. Would be better to adapt that instead of duplicating code.

Reply:
Or get dragon_integration_test.go to use this test cluster.

Reply:
In this PR I didn't want to also modify `dragon_integration_test` to use the test cluster. I mentioned it in this PR description. I can refactor `dragon_integration-test` and `dragon_snapshot_test` in a follow up PR.

Also `dragon_integration_test` is focused on other integrations. It's `TestMain` does not help. It's good to have integration tests with and without mTLS.

Reply:
> Also dragon_integration_test is focused on other integrations. It's TestMain does not help. It's good to have integration tests with and without mTLS.

We mainly use SQLTest for this

Reply:
> I can refactor dragon_integration-test and dragon_snapshot_test in a follow up PR.

Yes please do.

Reply:
(Aside: the dragon tests are somewhat vestigial, and there's an argument they're redundant given that we can run all SQL tests using different config)

Reply:
Agree on simplifying testing with TLS. I didn't know we had TLS tests in `sql_test`. Maybe it's enough to have a couple integration tests testing mTLS and have the other tests concentrate on what they're testing ...